src/main/java/Parser.java

```diff
@@ -0,0 +1,110 @@
+                return new DeleteCommand(taskNumber);
+            } else {
+                throw new DukeException("You need to include your task number to delete...");
+            }
+        } else if (inputInformation[0].equals(CommandType.TODO.getInput())) {
+            if (inputInformation.length > 1) {
+                String description = getStringFromArray(inputInformation, 1, inputInformation.length);
+                return new AddCommand(CommandType.TODO, description);
+            } else {
+                throw new DukeException("Your todo description can't be empty...");
+            }
+        } else if (inputInformation[0].equals(CommandType.DEADLINE.getInput())) {
+            if (inputInformation.length > 3) {
+                int indexOfBy = Arrays.asList(inputInformation).indexOf("/by");
+                if (indexOfBy == -1) {
+                    throw new DukeException("Did you include /by?");
+                } else if (indexOfBy == 1) {
+                    throw new DukeException("Did you include a description?");
+                } else {
+                    String description = getStringFromArray(inputInformation, 1, indexOfBy);
+                    String by = getStringFromArray(inputInformation, indexOfBy + 1, inputInformation.length);
+                    return new AddCommand(CommandType.DEADLINE, description, by);
+                }
+            } else {
+                throw new DukeException("Your deadline description or deadline can't be empty...");
+            }
+        } else if (inputInformation[0].equals(CommandType.EVENT.getInput())) {
+            if (inputInformation.length > 3) {
+                int indexOfAt = Arrays.asList(inputInformation).indexOf("/at");
+                if (indexOfAt == -1) {
+                    throw new DukeException("Did you include /at?");
+                } else if (indexOfAt == 1) {
+                    throw new DukeException("Did you include a description?");
+                } else {
+                    String description = getStringFromArray(inputInformation, 1, indexOfAt);
+                    String at = getStringFromArray(inputInformation, indexOfAt + 1, inputInformation.length);
+                    return new AddCommand(CommandType.EVENT, description, at);
+                }
+            } else {
+                throw new DukeException("Your event description or event period can't be empty...");
+            }
+        } else if (inputInformation[0].equals(CommandType.FIND.getInput())) {
+            if (inputInformation.length > 1) {
+                String searchDescription = getStringFromArray(inputInformation, 1, inputInformation.length);
+                return new FindCommand(searchDescription);
+            } else {
+                throw new DukeException("You need to include your keyword...");
+            }
+        } else {
+            throw new DukeException("My duck instincts tell me your input makes no sense...");
+        }
```

Comment:
Huge chunk of code here but because of great coding standards, they are understandable. Good job!

Reply:
I have refactored my code. Thanks for this review comment! 💯